File: Easy/Java/101. Symmetric Tree.java
class Solution {
    public boolean isSymmetric(TreeNode root) {
        return checkSymmetry(root, root);
    }

    public boolean checkSymmetry(TreeNode leftTree, TreeNode rightTree){
        if(leftTree == null && rightTree == null)
            return true;
        
        if(leftTree != null && rightTree != null && leftTree.val == rightTree.val){
            Boolean outer = checkSymmetry(leftTree.left, rightTree.right);
            Boolean inner = checkSymmetry(leftTree.right, rightTree.left);
            return outer && inner;
        }

        return false;
    }
}

File: Easy/Python/101. Symmetric Tree.py
class Solution(object):
    def checkSymmetry(self, leftTree, rightTree):
        if not leftTree and not rightTree:
            return True
        
        if leftTree and rightTree and leftTree.val == rightTree.val:
            outer = self.checkSymmetry(leftTree.left, rightTree.right)
            inner = self.checkSymmetry(leftTree.right, rightTree.left)
            return outer and inner
        
        return False
    
    def isSymmetric(self, root):
        return self.checkSymmetry(root, root)
